internal/lsp/lsproto/jsonrpc.go

```diff
@@ -207,7 +207,7 @@ func (r *RequestMessage) UnmarshalJSON(data []byte) error {
 type ResponseMessage struct {
 	JSONRPC JSONRPCVersion `json:"jsonrpc"`
 	ID      *ID            `json:"id,omitzero"`
-	Result  any            `json:"result"`
+	Result  any            `json:"result,omitzero"`
```

Comment:
```suggestion
	Result  *any           `json:"result,omitzero"`
```

I suspect that this would be needed to be able to still encode an explicit null, which many requests can respond with.

Reply:
Yeah, most are explicit like that, but I think some do need to explicitly return `null` like the `shutdown` request (the rest I think are server-to-client where we never send anything). Perhaps we could do your existing PR but just create some new explicit `null` value or something for that one case...

Reply:
I have a change which does the above (explicit null types for these cases); do you mind if I push it into this PR? Together I think the change would be correct.

Reply:
@jakebailey Of course, go ahead - that would be great.